src/test/llm/visual_language_model/test_complete_flow.cpp

```diff
@@ -0,0 +1,175 @@
+                        "url":  "data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAEElEQVR4nGIy+/oREAAA//8DiQIftNKCRwAAAABJRU5ErkJggg=="
+                        }
+                    }
+                ]
+            }
+            ]
+        )";
+    for (const auto& field : fields) {
+        oss << R"(, ")" << field.first << R"(": )" << field.second << R"()"
+            << "\n";
+    }
+    oss << "\n}";
+    return oss.str();
+}
+
+TEST_F(VLMServableFlowTest, unaryBasic) {
+    std::vector<std::pair<std::string, std::string>> fields = {
+        {"temperature", "0.0"},
+        {"stream", "false"},
+        {"max_tokens", "5"},
+        {"ignore_eos", "true"}};
+    std::string requestBody = createRequestBody(fields);
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::OK);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
+    int i = 0;
+    for (auto& choice : parsedResponse["choices"].GetArray()) {
+        ASSERT_TRUE(choice["finish_reason"].IsString());
+        EXPECT_STREQ(choice["finish_reason"].GetString(), "length");
+        ASSERT_EQ(choice["index"], i++);
+        ASSERT_FALSE(choice["logprobs"].IsObject());
+        ASSERT_TRUE(choice["message"].IsObject());
+        ASSERT_TRUE(choice["message"]["content"].IsString());
+        EXPECT_STREQ(choice["message"]["role"].GetString(), "assistant");
+    }
+
+    ASSERT_TRUE(parsedResponse["usage"].IsObject());
+    ASSERT_TRUE(parsedResponse["usage"].GetObject()["prompt_tokens"].IsInt());
+    ASSERT_TRUE(parsedResponse["usage"].GetObject()["completion_tokens"].IsInt());
+    ASSERT_TRUE(parsedResponse["usage"].GetObject()["total_tokens"].IsInt());
+    ASSERT_EQ(parsedResponse["usage"].GetObject()["completion_tokens"].GetInt(), 5 /* max_tokens */);
+    EXPECT_STREQ(parsedResponse["model"].GetString(), "dummyVLMServable");
+    EXPECT_STREQ(parsedResponse["object"].GetString(), "chat.completion");
+}
+
+TEST_F(VLMServableFlowTest, streamBasic) {
+    std::vector<std::pair<std::string, std::string>> fields = {
+        {"temperature", "0.0"},
+        {"stream", "true"},
+        {"max_tokens", "5"},
+        {"ignore_eos", "true"}};
+    std::string requestBody = createRequestBody(fields);
+
+    std::vector<std::string> responses;
+
+    EXPECT_CALL(*writer, PartialReply(::testing::_))
+        .WillRepeatedly([this, &responses](std::string response) {
+            responses.push_back(response);
+        });
+    EXPECT_CALL(*writer, PartialReplyEnd()).Times(1);
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::PARTIAL_END);
+    ASSERT_TRUE(responses.back().find("\"finish_reason\":\"length\"") != std::string::npos);
```

Comment:
how does this test work?

as far I remember PARTIAL_END is returned immediately and dispatchToProcessor is not blocking (http_rest_api_handler.cpp line 526)

it either means that we are lucky that it generates so fast that you have access to last message, or...?